src/parser.cpp

```diff
@@ -562,11 +562,14 @@ job_t *parser_t::job_get(job_id_t id) {
     return NULL;
 }
 
-job_t *parser_t::job_get_from_pid(int pid) {
+job_t *parser_t::job_get_from_pid(pid_t pid) {
     job_iterator_t jobs;
     job_t *job;
+
+    pid_t pgid = getpgid(pid);
+
     while ((job = jobs.next())) {
-        if (job->pgid == pid) return job;
+        if (job->pgid == pgid) return job;
```

Comment:
>No, the two processes should get the same PGID. All processes within a job have one job object and one PGID.

They should, but they currently don't in some cases - that's what causes #3952.

I was saying that this doesn't make those any worse and won't need to be changed again when it's fixed.